corehq/apps/locations/models.py

```diff
@@ -588,6 +581,37 @@ def unarchive(self):
         for loc in [self] + self.descendants:
             loc._unarchive_single_location()
 
+    def _close_case_and_remove_users(self):
+        """
+        Closes linked supply point cases for a location and unassigns the users
+        assigned to that location.
+
+        Used by both archive and delete methods
+        """
+
+        sp = self.linked_supply_point()
+        # sanity check that the supply point exists and is still open.
+        # this is important because if you archive a child, then try
+        # to archive the parent, we don't want to try to close again
+        if sp and not sp.closed:
+            close_case(sp.case_id, self.domain, COMMTRACK_USERNAME)
+
+        _unassign_users_from_location(self.domain, self._id)
+
+    def full_delete(self):
+        """
+        Delete a location and its dependants.
+        This also unassigns users assigned to the location.
+        """
+        to_delete = [loc for loc in [self] + self.descendants]
```

Comment:
Small, but self.descendants is already a list, so this could be `to_delete = [self] + self.descendants` (just as you do in line 581) without needing to iterate them here, and then again in line 611.